crates/hir/src/display.rs

```diff
@@ -596,33 +596,32 @@ impl HirDisplay for Trait {
         write_generic_params(def_id, f)?;
         write_where_clause(def_id, f)?;
 
-        let assoc_items = self.items(f.db);
-        let assoc_items_size = assoc_items.len();
-        let limited_size = f.limited_size.unwrap_or(assoc_items_size);
-        if assoc_items.is_empty() {
-            f.write_str(" {}")?;
-        } else {
-            f.write_str(" {\n")?;
-            for (index, item) in assoc_items.iter().enumerate() {
-                f.write_str("    ")?;
-                match item {
-                    AssocItem::Function(func) => {
-                        func.hir_fmt(f)?;
-                    }
-                    AssocItem::Const(cst) => {
-                        cst.hir_fmt(f)?;
-                    }
-                    AssocItem::TypeAlias(type_alias) => {
-                        type_alias.hir_fmt(f)?;
-                    }
-                };
-                f.write_str(",\n")?;
-                if index + 1 == limited_size && index + 1 != assoc_items_size {
-                    f.write_str("    ...\n")?;
-                    break;
+        if let Some(limit) = f.entity_limit {
+            let assoc_items = self.items(f.db);
+            let count = assoc_items.len().min(limit);
+            if count == 0 {
+                if assoc_items.is_empty() {
+                    f.write_str(" {}")?;
+                } else {
+                    f.write_str(" { /* … */ }")?;
                 }
+            } else {
+                f.write_str(" {\n")?;
+                for item in &assoc_items[..count] {
+                    f.write_str("    ")?;
+                    match item {
+                        AssocItem::Function(func) => func.hir_fmt(f),
+                        AssocItem::Const(cst) => cst.hir_fmt(f),
+                        AssocItem::TypeAlias(type_alias) => type_alias.hir_fmt(f),
+                    }?;
+                    f.write_str(";\n")?;
+                }
+
+                if assoc_items.len() > count {
+                    f.write_str("    /* … */\n")?;
+                }
+                f.write_str("}")?;
```

Comment:
@Young-Flash sorry for changing stuff myself here (but I figured this wasn't worth going through additional review rounds given how long the PR has been up).

Made some small stylistic changes (`;` instead of `,` for seperators, given items), and more importantly if no limit is set we don't show any items or braces at all. As before this we'd still see `{\n    ...\n}` for every trait hover which is rather noisy. If someone wants that they can set the limit to `0` instead of `null`.